dnscrypt-proxy/proxy.go

```diff
@@ -417,7 +415,6 @@ func (proxy *Proxy) processIncomingQuery(serverInfo *ServerInfo, clientProto str
 					sharedKey, encryptedQuery, clientNonce, err := proxy.Encrypt(serverInfo, query, serverProto)
 					if err != nil {
 						pluginsState.returnCode = PluginsReturnCodeParseError
-						pluginsState.ApplyLoggingPlugins(&proxy.pluginsGlobals)
```

Comment:
With the `defer`, that case may not be handled any more.

Reply:
Shouldn't it fail there?

Reply:
Actually, yes, it should fail :)